okdownload/src/main/java/com/liulishuo/okdownload/core/connection/DownloadUrlConnection.java

```diff
@@ -31,16 +34,46 @@
 public class DownloadUrlConnection implements DownloadConnection, DownloadConnection.Connected {
 
     protected URLConnection connection;
+    private Configuration configuration;
+    private URL url;
+    private RedirectHandler redirectHandler;
+
+    private static final String TAG = "DownloadUrlConnection";
 
     DownloadUrlConnection(URLConnection connection) {
+        this(connection, new RedirectHandler());
+    }
+
+    DownloadUrlConnection(URLConnection connection, RedirectHandler redirectHandler) {
         this.connection = connection;
+        this.url = connection.getURL();
+        this.redirectHandler = redirectHandler;
     }
 
     public DownloadUrlConnection(String originUrl, Configuration configuration) throws IOException {
         this(new URL(originUrl), configuration);
     }
 
     public DownloadUrlConnection(URL url, Configuration configuration) throws IOException {
+        this(url, configuration, new RedirectHandler());
+    }
+
+    public DownloadUrlConnection(
+            URL url,
+            Configuration configuration,
+            RedirectHandler redirectHandler) throws IOException {
+        this.configuration = configuration;
+        this.url = url;
+        this.redirectHandler = redirectHandler;
+        prepareConnection();
+    }
+
+    public DownloadUrlConnection(String originUrl) throws IOException {
+        this(originUrl, null);
+    }
+
+    void prepareConnection() throws IOException {
```

Comment:
I think this operation is better to be named as `configUrlConnection`  rather than `prepareConnection`.